### ISSUE TITLE
Final Proofreading: The AEGIS-128L Algorithm

### ISSUE THREAD

**Issue by samuel-lucas6** [cfrg/draft-irtf-cfrg-aegis-aead#75]
### Manual Review
https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3-4
`shall be` => `SHALL` or `MUST`

Should we add something about this in the Security Considerations?

From https://www.rfc-editor.org/rfc/rfc5116#section-4:
```
   Each AEAD algorithm MUST provide a description relating the length of
   the plaintext to that of the ciphertext.  This relation MUST NOT
   depend on external parameters, such as an authentication strength
   parameter (e.g., authentication tag length).  That sort of dependence
   would complicate the use of the algorithm by creating a situation in
   which the information from the AEAD registry was not sufficient to
   ensure interoperability.
```
We seem to be breaking this.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.1-3
Do we want to add to these `Security:` sections? They don't cover everything. They're more useful in the Decrypt() functions.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.1-10
Could change `for` to `foreach`.

`ai` could be `ad_block` and `xi` could be `msg_block`, but that's more verbose and overlaps with `ad_blocks`/`msg_blocks`.

`return ct and tag` is presumably so they can put the tag wherever. Otherwise, I'd do `ct || tag`.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.2-4.1.1
`message authentication tag` => `authentication tag`
and
`before being returned` => `before the function returns`

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.2-6.2.1
We could mention the length of the authentication tag, key, and nonce in these sections using the constants.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.2-10
I think it's also worth erasing the tag in all of these Decrypt() sections. The AEGIS paper says to do this, and we say to do this in the `Security:` subsection.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.6-8
https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.7-8
https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.8-8
`S6 ^ S1` could be swapped around to match the paper.

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.9-8
`# 256 bits` => alignment might want to be changed.

### Grammarly
https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3-2
`Section 4 are` => `Section 4, are`

https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-aegis-aead-14#section-3.2-6.1.1
`to be decrypted` => `to decrypt`

Otherwise, false positives or Pro suggestions that are unnecessary.

**Comment by jedisct1** [comment]
> Do we want to add to these Security: sections? They don't cover everything. They're more useful in the Decrypt() functions.

The fact that keys must be chosen from a random distribution and that nonces must not be reused apply to the `Encrypt` functions, not so much to the `Decrypt` functions. And they're the most important constraints to satisfy in order to use the `Encrypt` functions correctly, so yes, I think we should keep that.

**Comment by jedisct1** [comment]
> return ct and tag is presumably so they can put the tag wherever. Otherwise, I'd do ct || tag.

Yes, it's very common to return the tag and the ciphertext separately. This also allows in-place encryption/decryption. In `Stream`, we also discard the tag. Unconditionally returning a concatenation would make things more complicated.
